### PR TITLE
Examples with functional options

### PR DESCRIPTION
**Pull request by peterhellberg**
Use functional options for NewCollector in examples

### CODE REVIEW

**Comment by peterhellberg** [comment]
Travis-CI build failed due to connection problems between Travis-CI and GitHub when starting the `tip` build.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+
 go:
   - 1.7.x
   - 1.8.x
   - 1.9.x
   - tip
+
+sudo: false
+
+dist: trusty
```
Comment (asciimoo) [inline .travis.yml]: What's the benefit of explicitly declaring trusty? We have to manually update this if trusty becomes obsolete.

Comment (peterhellberg) [reply .travis.yml]: I thought it would be needed in order to force the use of the Container-based build environment, but maybe not.

Comment (asciimoo) [reply .travis.yml]: When I added `sudo: false` I noticed speedup immediately and if I compare the duration of this PR's travis job and the one where `sudo: false` was intrudoced I cannot see significant differences:
 - https://travis-ci.org/gocolly/colly/builds/325491037
 - https://travis-ci.org/gocolly/colly/builds/325611501

Comment (peterhellberg) [reply .travis.yml]: Then maybe I should remove that commit? (I also wonder why `"PATH=$GOPATH/bin:$PATH"` is set in the env section, `$GOPATH/bin` is automatically added to the $PATH by Travis-CI)

Comment (asciimoo) [reply .travis.yml]: You're right, then could you remove these lines?
If it's done, I'm merging this PR and migrating the new examples to the documentation site (I have a "beautiful" oneliner for this purpose =] ).

**Comment by peterhellberg** [comment]
I've replaced the commit changing `.travis.yml` with one that removes the env setting `"PATH=$GOPATH/bin:$PATH"`

**Pull request merged**